make/autoconf/lib-vmath.m4

```diff
@@ -0,0 +1,105 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+###############################################################################
+#
+# Setup vmath framework and check its dependences
+#
+AC_DEFUN_ONCE([LIB_SETUP_VMATH],
+[
+  AC_ARG_WITH(libsleef, [AS_HELP_STRING([--with-libsleef],
+      [specify prefix directory for the libsleef library
+      (expecting the libraries under PATH/lib and the headers under PATH/include)])])
+
+  LIBSLEEF_FOUND=no
+  LIBVMATH_CFLAGS=
+  LIBVMATH_LIBS=
+
+  if test "x${with_libsleef}" = xno; then
+    AC_MSG_NOTICE([libvmath will not be compiled, because its dependence libsleef is disabled in command line])
+  else
+    # Check the specified libsleef.so
+    if test "x${with_libsleef}" != x; then
+      AC_MSG_CHECKING([the specified LIBSLEEF])
+      if test -e ${with_libsleef}/lib/libsleef.so &&
+         test -e ${with_libsleef}/include/sleef.h; then
+        LIBSLEEF_FOUND=yes
+        LIBVMATH_LIBS="-L${with_libsleef}/lib"
+        LIBVMATH_CFLAGS="-I${with_libsleef}/include"
+      else
+        AC_MSG_ERROR([Could not locate libsleef.so or sleef.h in ${with_libsleef}])
+      fi
+      AC_MSG_RESULT([${LIBSLEEF_FOUND}])
+    fi
+
+    # Check the system locations if libsleef is not specified with option
+    if test "x$SYSROOT" = x && test "x${LIBSLEEF_FOUND}" = "xno"; then
+      PKG_CHECK_MODULES([LIBSLEEF], [sleef], [LIBSLEEF_FOUND=yes], [LIBSLEEF_FOUND=no])
+    fi
+    if test "x$LIBSLEEF_FOUND" = xno; then
+      AC_CHECK_HEADERS([sleef.h],
+          [LIBSLEEF_FOUND=yes],
+          []
+      )
+    fi
+
+    if test "x${LIBSLEEF_FOUND}" = "xyes"; then
+      LIBVMATH_LIBS="${LIBVMATH_LIBS} -lsleef"
+
+      if test "x${OPENJDK_TARGET_CPU}" = "xaarch64"; then
+        # Check the ARM SVE feature
+        SVE_FEATURE_SUPPORT=no
+        SVE_CFLAGS="-march=armv8-a+sve"
+
+        AC_LANG_PUSH(C)
+        OLD_CFLAGS="$CFLAGS"
+        CFLAGS="$CFLAGS $SVE_CFLAGS"
+
+        AC_MSG_CHECKING([if ARM SVE feature is supported])
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <arm_sve.h>],
+            [
+              svint32_t r = svdup_n_s32(1);
+              return 0;
+            ])],
+            [
+              SVE_FEATURE_SUPPORT=yes
+              LIBVMATH_CFLAGS="${LIBVMATH_CFLAGS} ${SVE_CFLAGS}"
+            ],
+            []
+        )
+        AC_MSG_RESULT([${SVE_FEATURE_SUPPORT}])
```

Comment:
What is this test even for? I can't see any usage of SVE_FEATURE_SUPPORT outside this function.

Reply:
This is just used to print the result of `AC_MSG_CEHCKING[if ARM SVE feature is supported]` in configure.

Reply:
Ah, now I se what you are trying to do here. First of all, in the detection part, only set `SVE_FEATURE_SUPPORT`. Then you can handle the `SVE_CFLAGS` addition elsewhere/later. 

Secondly, you should not mix these `SVE_CFLAGS` with the spleef C flags. Keeping them separate will allow for LIBSLEEF_CFLAGS to be named just that.

Thirdly, I do not like at all how you just come crashing in setting `-march` like that. The `-march` flag is handled by `FLAGS_SETUP_ABI_PROFILE`. 

Actually, now that I think of it, this is just completely wrong! You are checking on features on the build machine, to determine what target machine code to generate, with no way to override. 

You need to break out the -march handling separately. It should be moved to FLAGS_SETUP_ABI_PROFILE. I'm guessing you will need to make something like a `aarch64-sve` profile, and possibly try to auto-select it based on the result of the sve test program above. But changing `OPENJDK_TARGET_ABI_PROFILE` can have further consequences; I do not know the full extent on the top of my head.

Reply:
Thanks for the advice! I will take a consideration for it.

Reply:
> Thirdly, I do not like at all how you just come crashing in setting -march like that. The -march flag is handled by FLAGS_SETUP_ABI_PROFILE.

`-march=armv8-a+sve` is just used in this new added module, which may not expect to be used for other libraries. Per my understanding, flags handled by `FLAGS_SETUP_ABI_PROFILE` is not used for a specified module? 

> Actually, now that I think of it, this is just completely wrong! You are checking on features on the build machine, to determine what target machine code to generate, with no way to override.

Yes, that's be a risk, although the usage to the SVE functions are controlled by SVE feature as well in runtime. I need time to find a better solution.

Reply:
It does not matter if you set the -march on just part of the build. Actually, there is no point in doing so. Either the JDK is run on a machine with the matching architecture, or it isn't. 

I don't know the details of what the aarch64 SVE feature means, but unless this is a special instance, any attempt to execute the compiled code on a machine that does not support that architecture will fail.